test/quantization/test_quantize_fx.py

```diff
@@ -102,15 +112,74 @@ def forward(self, x):
             e = qgraph_script(*inputs)
             e_debug = qgraph_debug(*inputs)
 
-            found = False
-            modules = dict(qgraph.root.named_modules())
-            for node in qgraph.graph.nodes:
-                if node.op == 'call_function':
-                    found = found or node.op == quantized_node[0] and node.target == quantized_node[1]
-                elif node.op == 'call_module':
-                    found = found or node.op == quantized_node[0] and type(modules[node.target]) == quantized_node[1]
-            assert found, 'Expected to find quantized node:' + str(quantized_op)
-            # assert (a-d).abs().max() < 2
+            self.checkGraphModuleHasNode(qgraph, quantized_node)
             assert torch.allclose(d, e)
             assert (d - d_debug).abs().max() == 0
             assert (e - e_debug).abs().max() == 0
+
+
+class TestQuantizeFxOps(QuantizationTestCase):
+    """Unit tests for individual ops
+    """
+    @unittest.skip("Enable after tensor is supported in fx")
+    def test_linear(self):
+        class ModuleLinear(torch.nn.Module):
+            def __init__(self, has_relu=False, f_relu=False):
+                super(ModuleLinear, self).__init__()
+                self.linear = torch.nn.Linear(30, 4).float()
+                if has_relu:
+                    if f_relu:
+                        self.relu = F.relu
+                    else:
+                        self.relu = torch.nn.ReLU()
+                else:
+                    self.relu = torch.nn.Identity()
+
+            def forward(self, x):
+                return self.relu(self.linear(x))
+
+        class FuncLinear(torch.nn.Module):
+            def __init__(self, has_relu=False, f_relu=False):
+                super(FuncLinear, self).__init__()
+                self.w = torch.randn(4, 30)
+                self.b = torch.randn(4)
+                if has_relu:
+                    if f_relu:
+                        self.relu = F.relu
+                    else:
+                        self.relu = torch.nn.ReLU()
+                else:
+                    self.relu = torch.nn.Identity()
+
+            def forward(self, x):
+                return self.relu(F.linear(x, self.w, self.b))
+
+        data = [torch.rand((1, 30), dtype=torch.float)]
+        options = itertools.product(
+            [(ModuleLinear(has_relu=False), True),
+             (FuncLinear(has_relu=False), False)],
```

Comment:
are we missing a check for `has_relu=True`?

Reply:
we do have it in L181, although some of the functionalities(functional linear + relu) are not supported yet

Reply:
I see, I think we also need to add linear dynamic + relu fusion as well.